### DISCUSSION TITLE
[New] Proposition d'ajout de pratique : « Utiliser le chargement paresseux des compartiments »

### DISCUSSION THREAD

**Discussion by florinesueur** [cnumr/best-practices#30] [Bonnes pratiques] [answered]
A la maniière des images, il est possible, par exemple, pour le navigateur de ne pas travailler sur le rendu d'articles qui ne sont pas dans la zone visible.

Lien MDN : [https://developer.mozilla.org/fr/docs/Web/CSS/CSS_Containment](https://developer.mozilla.org/fr/docs/Web/CSS/CSS_Containment)

**Comment by DocRoms** [comment]
Très bonne idée, surtout lorsque le site possède beaucoup (trop) d'élément de DOM, ou beaucoup (trop) de CSS. 
    - Lien vers la doc officielle (w3.org) ➡️  https://www.w3.org/TR/css-contain-2/
    - Lien vers une explication intéressante de procédé ➡️ https://blogs.igalia.com/mrego/2019/01/11/an-introduction-to-css-containment/

**Comment by ACTLEM** [comment]
Bonne idée, j'ai juste un questionnement sur la compatibilité. J'ai l'impression par exemple que sur iOS ce n'est pas encore supporté (Safari et Chrome), mais je me trompe peut-être.

**Reply by florinesueur** [reply]
En effet, ce n'est pas encore pris en charge par Safari. https://caniuse.com/mdn-css_properties_contain

**Comment by ACTLEM** [comment, marked as answer]
https://github.com/cnumr/best-practices/issues/134

**Comment by Christophe-Cl** [comment]
Attention, le "chargement paresseux" est souvent confondu avec le "lazy-loading", donc différent des "ressources uniquement dédiées à l'affichage de ce qui est visible sur l'écran de l'utilisateur".
Il faut certainement distinguer les 2 : le calcul uniquement nécessaire/prioritaire aux contenus visibles (afin d'éliminer par ex. les calculs JS en scroll inférieur et prioriser la structure de rendu au DOM), et la méthode le "lazy-loading" de contenu.

À noter que le "lazy-loading" n'est pas forcément une BP universelle qui fait consensus. Par exemple, un accès raccourci type ancre-html situé en haut de page vers un pied de page peut charger l'intégralité des ressources présentées en lazy-load, alors qu'il peut totalement les éviter (donc ne pas les charger) si il n'y a pas la présence de lazy-load.
